### ISSUE TITLE
Default collection item id for single type collections.

### ISSUE THREAD

**Issue by tspvivek** [directus/app#2991]
This is the best headless CMS with almost all of the features needed for developing applications. Problems are mostly with single type collections.

1. When we open a single type collection for the first time, it tries to open the item with id 1. But the item will not be available for newly created collection, so there should be option to add item when its opened first time.

2. When a single type collection is given "mine only" read permission it should open the first item of the logged in user instead of the item id 1.
( I know its hard coded in vue js with item id 1, if there is any option like passing item id as "single" or something else from vue js and gets the first item of the user based on that, like "https://directus.url/myproject/items/mycollection/single" )

3. In a directus installation with multiple projects, in login screen there should be option to hide the dropdown and show input field to input the project name. Because dropdown reveals all the available projects to the users.

**Comment by rijkvanzanten** [comment]
Not sure if I agree with

> When a single type collection is given "mine only" read permission it should open the first item of the logged in user instead of the item id 1.

Personally, there's a big difference between having a "true" singleton (like "app settings", or "homepage") where everybody edits the same record, versus the type of setup you describe where every user is limited to only 1 row 🤔

**Comment by tspvivek** [comment]
@rijkvanzanten Hi, The use case is, We are using directus to build SaaS system with multiple users managing their own website. So each user has his own website settings.

**Comment by rijkvanzanten** [comment]
Yeah! I understand the need, I'm just saying that I don't want it to replace the way it currently works 🙂

**Comment by tspvivek** [comment]
@rijkvanzanten Is there any other way we can achieve this? like using custom hooks. We almost completed the development but got stuck here.

**Comment by benhaynes** [comment]
Any updates to v8 will be from the community, as all of our resources are going to v9 from now on.

Number 3 doesn't seem related to Singletons... and it doesn't apply to v9, since the new version will only manage one project. We'll likely be creating a super-admin layer on top of DIrectus proper, which will act as a layer for managing multiple project instances.

I agree with what Rijk said for number 2, that's an interesting idea, but breaks from how we've defined a singleton. I would say you can open a Feature Request within GitHub Discussions (again, for v9) to see how best to create something like this (although a normal collection with permissions gets you 80% there... just not skipping the listing page).

**Comment by tspvivek** [comment]
@rijkvanzanten @benhaynes  After few tries we figured out a way in hooks, by overriding the input item id 1 with the item id of the logged in user (fetched from DB directly) in update filter and also insert one if there is no entry for the user in read filter (to fix the first one). Hope this will work after the coding is done.
